### ISSUE TITLE
relic_pc: gt_is_unity() does not work as expected

### ISSUE THREAD

**Issue by 1gn1t10n** [relic-toolkit/relic#8]
#### Context
RELIC version: 1d085285a37f75bdbddf1c1c7dce7af3d0bd4d3e
Configure parameters: `-DDEBUG=on -DBENCH=10 -DTESTS=10 -DARITH=gmp -DALLOC=DYNAMIC -DBN_PRECI=8192 -DFP_PRIME=256`

For this configuration, `gt` is `fp12`.

##### Example 1
```
fp12_print(a)
1B6EEEBC330A20B5 8B3F6CCCC5EA3A6B 160B8CE4BA2AD063 9C8B8428CD3000EF
AD57C9EDDD332E0B D7C21344C757BE6E AA81013EDE3AC7E9 9CED22772A962FC2
66BDABD9ECCE14E6 C6CF36B0B4E4F66B F5CB6968D0250A99 59021F08FC7D3A9E
6BFE7B284667E5D1 C57A79D2C1691F2B 54A802447372D6A9 998BB60F1797B5DC
AA23F7E21D98F25A B1E81A80EAAAA7D7 6225AD45329CD85F F6FB647C2811C02B
58F1FD7BC67E50AA 31130AF193C80A48 1BA53A61F1345CEF 38D039EA0B639A51
3D2442CC297C1AC2 F91AD601C8F207F8 C996BE68ABE56316 0690C2D8C9A109FC
B5B0F937BDBC4BE1 BACE3D838C73051A DA1DAB1AE4AAFDC8 B616DD52993EEFD6
A04EEF6A9DC425A5 77DECA0ACA80C295 73EB12826706DB3D F3D3778A4979CCD9
5476C9853F92B5AA 6A7D1752B1162FE8 DF7AC7E377AD0EAA 6CB70D9713F5921E
5145CB73ABA6F97C 64E500488E6BAE1A 67081B38651102FE 10A4C7442EB45DEF
62B4000B57990329 DF2EB2C936D2986C 63822B65193A62E1 B22FBD975D7C718E
```
Testing `if(gt_is_unity(key))` always evaluates to true.

##### Example 2
The following addition to the test suite of the `pc` module highlights the issue:
``` diff
diff --git a/test/test_pc.c b/test/test_pc.c
index 041d1b2..c3e53ab 100755
--- a/test/test_pc.c
+++ b/test/test_pc.c
@@ -1081,6 +1081,12 @@ int util(void) {
                        TEST_ASSERT(gt_is_unity(a), end);
                }
                TEST_END;
+
+//           TEST_BEGIN("EXTRA unity test") {
+//                   gt_rand(a);
+//                   TEST_ASSERT(!gt_is_unity(a), end);
+//           }
+//           TEST_END;
        }
        CATCH_ANY {
                util_print("FATAL ERROR!\n");
```

With the bug, the test should fail with high probability.

##### Developer side
A manual call to `fp12_cmp_dig(key, 1) == CMP_EQ` works.
More generically, setting an element to `gt_set_unity(unity)` and then comparing `gt_cmp(key, unity) == CMP_EQ` works.

##### Library side
`gt_is_unity`  is a macro that expands to `fp12_cmp_dig (a=0xxxx, b=1)` (or `fp2_cmp_dig` for the supersingular curve). The result of the comparison (be it fp12_cmp or fp2_cmp) should be checked against CMP_EQ.

**Comment by dfaranha** [comment]
I believe the code below fixes the issue, while keeping the interface clear:

#define gt_is_unity(P)		(CAT(GT_LOWER, cmp_dig)(P, 1) == CMP_EQ)

**Comment by 1gn1t10n** [comment]
Yes, the above modification available in commit fa48d4535595096d6810243e4f912f9f75ca71dd fixes the issue. Thank you!
